packages/extension-chakra-storefront/src/components/offline-boundary/index.test.js

```diff
@@ -40,84 +40,70 @@ describe('The OfflineBoundary', () => {
+        expect(screen.queryByText(/child/i)).not.toBeInTheDocument()
+    })
 
-    // TODO: Fix flaky/broken test
-    // eslint-disable-next-line jest/no-commented-out-tests
-    // test('should re-throw errors that are not chunk load errors', () => {
-    //     const ThrowingComponent = () => {
-    //         throw new Error('Anything else')
-    //     }
-    //     expect(() => {
-    //         renderWithRouter(
-    //             <OfflineBoundary isOnline={true}>
-    //                 <div>
-    //                     <ThrowingComponent />
-    //                     <div id="child">child</div>
-    //                 </div>
-    //             </OfflineBoundary>
-    //         )
-    //     }).toThrow()
-    // })
+    test('should re-throw errors that are not chunk load errors', () => {
+        const ThrowingComponent = () => {
+            throw new Error('Anything else')
+        }
+        expect(() => {
+            renderWithProviders(
+                <OfflineBoundary isOnline={true}>
+                    <div>
+                        <ThrowingComponent />
+                        <div id="child">child</div>
+                    </div>
+                </OfflineBoundary>
+            )
+        }).toThrow()
+    })
 
-    // TODO: Fix flaky/broken test
-    // eslint-disable-next-line jest/no-commented-out-tests
-    // test('should attempt to reload the page when the user clicks retry', () => {
-    //     let firstRender = true
-    //     const ThrowingOnceComponent = () => {
-    //         if (firstRender) {
-    //             firstRender = false
-    //             throw new ChunkLoadError()
-    //         } else {
-    //             return <div id="child">child</div>
-    //         }
-    //     }
-    //     renderWithRouter(
-    //         <OfflineBoundary isOnline={true}>
-    //             <ThrowingOnceComponent />
-    //         </OfflineBoundary>
-    //     )
+    test('should call clearError when retry button is clicked', async () => {
+        const user = userEvent.setup()
+        const ThrowingComponent = () => {
+            throw new ChunkLoadError()
+        }
+        const clearErrorSpy = jest.spyOn(UnwrappedOfflineBoundary.prototype, 'clearError')
 
-    //     expect(screen.getByRole('img', {name: /offline cloud/i})).toBeInTheDocument()
-    //     expect(
-    //         screen.getByRole('heading', {name: /you are currently offline/i})
-    //     ).toBeInTheDocument()
-    //     expect(screen.queryByText(/child/i)).not.toBeInTheDocument()
+        renderWithProviders(
+            <OfflineBoundary isOnline={true}>
+                <ThrowingComponent />
+            </OfflineBoundary>
+        )
+
+        expect(screen.getByRole('img', {hidden: true})).toBeInTheDocument()
+        expect(
+            screen.getByRole('heading', {name: /you are currently offline/i})
+        ).toBeInTheDocument()
+
+        await user.click(screen.getByRole('button', {name: /retry connection/i}))
```

Comment:
Let's wrap user event inside `act` to avoid React warnings in the terminal. 
```
import {act} from 'testing-library/react'

await act(() => {
        await user.click(screen.getByRole('button', {name: /retry connection/i}))
})

```